### ISSUE TITLE
Initial price of Warriors at the Bazaar

### ISSUE THREAD

**Issue by pauljtaylor** [ratsputin/Dark-Tower#1]
According to the comments at line 3737 and 3751, the initial price of a Warrior is 3 to 8 gold.

The code seems to indicate the following:
3753: A number between 0 to 15 is generated
3757: That number gets converted to 0 to 7
3758: The number again gets converted to 0 to 3
3767: Add 5 to the number,

If the price is then (0 to 3) + 5, isn't the resulting range 5 to 8?

**Comment by ratsputin** [comment]
You are absolutely correct.  I'm not really sure what I was thinking at the time.  Thank you for catching that; I'll get it updated.

**Comment by pauljtaylor** [comment]
Also, I've got a thread over at BoardGameGeek pouring over the findings:
https://boardgamegeek.com/thread/1087086/article/39778168
(I'm Gazoxtahagen there.)
There's a couple of passages that you don't comment on, namely how the Tower determines the number of Brigands at the start of a battle (although I believe through a deep dive into L800 that it's a random number between 0 and 2 added to or subtracted from the Warrior count), and also the odds of a Warrior victory given a number of Warriors X over a number of Brigands Y.  Any additional insight that you could add as code comments would be much appreciated; these are mysteries that the gaming community has been pouring over for decades!

**Comment by pauljtaylor** [comment]
Also, a similar issue with haggling.  The code on line 3942 indicates that a success roll of "0-11" is a "68.75%" chance of success.  But that's on a roll of 0-15.  Isn't that a 75% chance of success?  It looks like the top of the roll is #A, which is decimal 10, not 11.  So my guess is that your odds of 68.75% are correct and that comment that the range is 0-11 that should be corrected to 0-10.

**Comment by ratsputin** [comment]
Well, it's been almost four years since I've circled back and looked at the Dark Tower code.  Better late than never, huh?

With the help of Google Gemini and the MAME debugger I was able to get the code at L800 documented as well as a number of corrections and updates to the Excel spreadsheet.

I plan to start focusing on the Bazaar code next.  It's pretty gnarly.
